src/kafka-util/src/client.rs

```diff
@@ -70,20 +73,21 @@ impl ProducerContext for MzClientContext {
 }
 
 /// A client context that supports rewriting broker addresses.
+#[derive(Clone)]
 pub struct BrokerRewritingClientContext<C> {
-    inner: C,
+    inner: Arc<C>,
     overrides: BTreeMap<BrokerAddr, BrokerAddr>,
     /// Opaque tokens to cleanup resources associated with overrides.
-    drop_tokens: Vec<Box<dyn Any + Send + Sync>>,
+    drop_tokens: Arc<Mutex<Vec<Box<dyn Any + Send + Sync>>>>,
```

Comment:
Why did these change to `Arc`s and `Mutex`s?

Reply:
The context has to be `Clone`-able to make `create_with_context` able to retry